pkg/policy/common.go

```diff
@@ -91,25 +91,36 @@ func ExcludePod(resourceMap map[string]unstructured.Unstructured, log logr.Logge
 	return resourceMap
 }
 
-// GetNamespacesForRule gets the matched namespaces list for the given rule
-func GetNamespacesForRule(rule *kyverno.Rule, nslister listerv1.NamespaceLister, log logr.Logger) []string {
+// getNamespacesForRule gets the matched namespaces list for the given rule
+func (pc *PolicyController) getNamespacesForRule(rule *kyverno.Rule, log logr.Logger) []string {
+	var results []string
+	var matchedNS []string
+
+	defer func() {
+		for _, ns := range matchedNS {
+			if !pc.configHandler.ToFilter("", ns, "") {
+				results = append(results, ns)
+			}
+		}
+	}()
```

Comment:
Can we move this to a utility method `FilterNamespaces` in the config package? Seems like we are probably doing the same logic in different modules....

Reply:
`ToFilter` is used to filter by `namespace` or `kind` or `name`, I can add `FilterNamespaces` which takes multiple namespaces and return allowed namespaces.